### PR TITLE
migrate serviceaccount handling to the dynamic client

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Migrate ServiceAccount handling to the dynamic client

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
E2E tests will succeed even if this doens't work, right?

File: api/pkg/controller/monitoring/resources.go [hunk 1]
```diff
@@ -176,3 +162,16 @@ func (c *Controller) ensureServices(cluster *kubermaticv1.Cluster, data *resourc
 
 	return resources.ReconcileServices(creators, cluster.Status.NamespaceName, c.dynamicClient, c.dynamicCache, resources.ClusterRefWrapper(cluster))
 }
+
+// GetServiceCreators returns all service creators that are currently in use
+func GetServiceAccountCreators() []resources.ServiceAccountCreator {
```
Comment (alvaroaleman) [inline api/pkg/controller/monitoring/resources.go:167]: I personally would prefer it if we used `NamedRESOURCECreatorGetter` for everything btw, whats your take on that?

**Comment by mrIncompetent** [comment]
> E2E tests will succeed even if this doens't work, right?

Yes, we don't have any test which validates Prometheus

File: api/pkg/controller/monitoring/resources.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/monitoring/resources.go:167]: Yes, i agree, but i would like to issue a dedicated PR to switch all resources to the `NamedRESOURCECreatorGetter`.
I first would like to remove the need for all custom `Ensure***` functions to not mix different refactoring steps

**Comment by alvaroaleman** [comment]
Lets merge this before #2882 as this makes the latter less ugly

**Pull request merged**